openlibrary/plugins/openlibrary/api.py

```diff
@@ -428,21 +429,49 @@ class observations(delegate.page):
     path = "/observations"
     encoding = "json"
 
+    def GET(self):
+        return delegate.RawText(json.dumps(get_observations()), content_type="application/json")
+
     def POST(self):
         user = accounts.get_current_user()
 
-        if user:
-            account = OpenLibraryAccount.get_by_email(user.email)
-            s3_keys = web.ctx.site.store.get(account._key).get('s3_keys')
+        if not user:
+            raise web.seeother('/account/login')
+
+        data = json.loads(web.data())
+        work_id = int(extract_numeric_id_from_olid(data['work_id']))
+
+        Observations.persist_observations(
+            data['username'],
+            work_id,
+            data['observations']
+        )
+
+        def response(msg, status="success"):
+            return delegate.RawText(json.dumps({
+                status: msg
+            }), content_type="application/json")
 
-            if s3_keys:
-                response = post_observation(web.data(), s3_keys)
-                return delegate.RawText(response)
+        return response('Observations added')
```

Comment:
This should be moved to the `patron_observations` class.  POSTing to `/observations` should update observations types and values.